### DISCUSSION TITLE
Timer class to get callback called

### DISCUSSION THREAD

**Discussion by markusgritsch** [lona-web-org/lona#399] [Feature Request]
If I want to periodically update some Browser content, I need to add an endless loop at the end of  `handle_request()` like this:
```python
import time
from lona import LonaApp, LonaView

app = LonaApp(__file__)

@app.route('/')
class MyLonaView(LonaView):
    def handle_request(self, request):
        while True:
            self.show(int(time.time()))
            self.sleep(1)

app.run()
```

When updating different things at different intervals, the necessary manual dispatching quickly becomes cumbersome.

From Qt I'm used to be able to us a [QTimer](https://doc.qt.io/qt-6/qtimer.html) object to call a callback method, either periodically, or just one time in single-shot mode.

What do you think about such a class for Lona?
NiceGUI also seems to have something similar: https://nicegui.io/documentation/timer

**Comment by fscherf** [comment]
Great suggestion. In fact, I ran into the same problem, with one of my projects. I had an idea for timers too but with a different approach.
I am currently working on channels for Lona (https://github.com/lona-web-org/lona/issues/299). The idea is to have a simple publish-subscribe system, that works between views, middlewares and non-Lona code like Django db-signals for example.
The channels API is intended to replace Lona view-events.

My idea is to have generic timer topics, that work like cron-jobs, you can subscribe to.

```python
from lona import Channel

def my_update_code(channel, message):
    # gets called every 30sec
    pass

# format: "timer.{hour}.{minute}.{second}"
timer = Channel('timer.*:*:/30', my_update_code)
```
